sdk/src/signer.rs

```diff
@@ -129,6 +129,10 @@ pub trait AsyncSigner: Sync {
         None
     }
 
+    fn timestamp_request_body(&self, message: &[u8]) -> Result<Vec<u8>> {
```

Comment:
Changes look good.  Where is this used?

Reply:
Ideally it would be used in the send_timestamp_request() implemenation as below. Howerver the current implemenation has a check to compare the nonce from the request and response as an extra step of validation.  With current API that would require parsing the body back again. Is that OK @mauricefisher64 or can we ignore the check?
```rust
    async fn send_timestamp_request(&self, message: &[u8]) -> Option<Result<Vec<u8>>> {
        let headers: Option<Vec<(String, String)>> = self.timestamp_request_headers();

        if let (Some(url), Some(body)) == (self.time_authority_url(), self.timestamp_request_body(message)) {
           let body = body?;
           // send http request. with headers and body.
       }
    }

```

Reply:
Actually fixed this by parsing the header bytes back to TimeStampReq and validating the nonce.